components/TwitterTestimonials.vue

```diff
@@ -0,0 +1,118 @@
+<template>
+  <v-card
+    class="mx-auto tweet-widget"
+    :class="$vuetify.breakpoint.mobile ? 'mobile' : ''"
+    color="white"
+  >
+    <v-carousel
+      v-model="activeTweetIndex"
+      hide-delimiters
+      :show-arrows="!$vuetify.breakpoint.mobile"
+      light
+    >
+      <v-carousel-item v-for="(tweet, i) in tweets" :key="tweet.id">
+        <tweet :tweet="tweets[i]" />
+      </v-carousel-item>
+    </v-carousel>
+    <tweet-carousel-delimiter
+      v-show="$vuetify.breakpoint.mobile"
+      :steps="3"
+      :index="activeTweetIndex"
+    />
+  </v-card>
+</template>
+<script>
+import testimonialsData from "../data/testimonials.json";
+import tweetFixtures from "../fixtures/twitterTweetsApiFixture.json";
+
+const API_URL =
+  "https://api.twitter.com/2/tweets?user.fields=profile_image_url,username&expansions=author_id&ids=";
+const DEV_MODE = "DEV_MODE";
+
+const token = process.env.TWITTER_API_BEARER_TOKEN || DEV_MODE;
+const options = {
+  headers: { Authorization: `Bearer ${token}` },
+  mode: "no-cors",
+};
+
+export default {
+  data() {
+    return {
+      tweets: [],
+      activeTweetIndex: 0,
+      isLoaded: false,
+    };
+  },
+  methods: {
+    createUserMap: function (apiUserData) {
+      /* Maps users by id */
+      const userMap = {};
+      for (const user of apiUserData) {
+        userMap[user.id] = user;
+      }
+      return userMap;
+    },
+    annotateTweetsWithUsers: function (apiTweetData, userMap) {
+      /* The twitter API separates user information and tweet content. This method recombines, overwriting fields as necessary */
+      let annotated = [];
+      for (const tweet of apiTweetData) {
+        const annotatedTweet = Object.assign(tweet, {
+          name: userMap[tweet.author_id].name,
+          username: userMap[tweet.author_id].username,
+          profile_image_url: userMap[tweet.author_id].profile_image_url,
+          profile_image_url_original: userMap[
+            tweet.author_id
+          ].profile_image_url.replace(/_normal\./, "."),
+        });
+        annotated.push(annotatedTweet);
+      }
+      return annotated;
+    },
+  },
+  async fetch() {
```

Comment:
method name is same as a reserved `fetch`. maybe `fetchTweets` or `fetchTestimonials`?

Reply:
This is actually the recommended pattern from the [Nuxt data fetching documentation](https://nuxtjs.org/docs/2.x/features/data-fetching). One of these is a component-level fetch hook, and the other is the call to the http service. I also think it's a bit confusing that they did this, and while we could either revise to a different name, then put a hook elsewhere, or call the http call as `$http` instead of `fetch`, but neither is particularly elegant. I'd stick to the official pattern here unless we have a reason to change.

Reply:
Ah, after reading the doc, you're right. 👍 

side-note:
it'ssuper sketchy that they call `fetch() {  ...await fetch('') }`. (just checked out the doc, since I've typically used straight-up axios) Is this only working based on the fact that the parent/vue method fetch doesn't have a parameter?

Reply:
Yeah, I can't say I love it, but if there's a canonical pattern I generally think it should be followed when using a framework, so I'll keep this. But yeah, why it doesn't run into namespacing issues is ???.